### PR TITLE
SCB-151 Update to add the pack-demo

### PR DESCRIPTION
**Pull request by zhfeng**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: saga-demo/pack-demo/booking/src/main/java/org/apache/servicecomb/saga/demo/pack/booking/Application.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package org.apache.servicecomb.saga.demo.pack.booking;
+
+import org.apache.servicecomb.saga.omega.spring.EnableOmega;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+/**
+ * @author <a href="mailto:zh.feng@google.com>Zheng Feng</a>
```
Comment (WillemJiang) [inline saga-demo/pack-demo/booking/src/main/java/org/apache/servicecomb/saga/demo/pack/booking/Application.java]: We normally don't use the author here, as the user could ask question in the community mailing list.

File: saga-demo/pack-demo/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## Running Demo
+1. run the following command to create docker images in saga project root folder.
+```
+mvn package -DskipTests -Pdocker -Pdemo
+```
+
+2. start application up in saga/saga-demo/pack-demo with the following command
+```
+docker-compose create
+docker-compose start
```
Comment (eric-lee-ltk) [inline saga-demo/pack-demo/README.md]: `docker-compose up` is same as the above two commands.

File: saga-demo/pack-demo/booking/pom.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      <groupId>org.apache.servicecomb.saga</groupId>
+      <artifactId>omega-spring-starter</artifactId>
+    </dependency>
+    <dependency>
+       <groupId>org.apache.servicecomb.saga</groupId>
+       <artifactId>omega-transport-resttemplate</artifactId>
+     </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-test</artifactId>
```
Comment (eric-lee-ltk) [inline saga-demo/pack-demo/booking/pom.xml]: the pom of `saga-demo` has already included this dependency~

File: saga-demo/pack-demo/car/pom.xml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      <groupId>org.apache.servicecomb.saga</groupId>
+      <artifactId>omega-spring-starter</artifactId>
+    </dependency>
+    <dependency>
+       <groupId>org.apache.servicecomb.saga</groupId>
+       <artifactId>omega-transport-resttemplate</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-test</artifactId>
```
Comment (eric-lee-ltk) [inline saga-demo/pack-demo/car/pom.xml]: same as above

File: saga-demo/pack-demo/hotel/pom.xml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      <groupId>org.apache.servicecomb.saga</groupId>
+      <artifactId>omega-spring-starter</artifactId>
+    </dependency>
+    <dependency>
+       <groupId>org.apache.servicecomb.saga</groupId>
+       <artifactId>omega-transport-resttemplate</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-test</artifactId>
```
Comment (eric-lee-ltk) [inline saga-demo/pack-demo/hotel/pom.xml]: same as above

File: saga-demo/pack-demo/hotel/src/main/java/org/apache/servicecomb/saga/demo/pack/hotel/HotelBookingController.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  @Autowired
+  private HotelBookingService service;
+
+  private AtomicInteger id = new AtomicInteger(0);
+
+  @GetMapping("/bookings")
+  List<HotelBooking> getAll() {
+    List<HotelBooking> result = new ArrayList<>();
+    result.addAll(service.getAllBookings());
+    return result;
```
Comment (eric-lee-ltk) [inline saga-demo/pack-demo/hotel/src/main/java/org/apache/servicecomb/saga/demo/pack/hotel/HotelBookingController.java]: the above can be simplified in one line as `return new ArrayList<>(service.getAllBookings());`

File: saga-demo/pack-demo/car/src/main/java/org/apache/servicecomb/saga/demo/pack/car/CarBookingController.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+public class CarBookingController {
+  @Autowired
+  private CarBookingService service;
+
+  private AtomicInteger id = new AtomicInteger(0);
+
+  @GetMapping("/bookings") List<CarBooking> getAll() {
+    List<CarBooking> result = new ArrayList<>();
+    result.addAll(service.getAllBookings());
+    return result;
```
Comment (eric-lee-ltk) [inline saga-demo/pack-demo/car/src/main/java/org/apache/servicecomb/saga/demo/pack/car/CarBookingController.java]: the above can be simplified in one line as `return new ArrayList<>(service.getAllBookings());`

**Comment by zhfeng** [comment]
Thanks all for your reviews and I'd update it with the suggestions.

**Pull request merged**